api/v1beta1/common_types.go

```diff
@@ -116,6 +111,32 @@ type WatcherTemplate struct {
 	// +kubebuilder:default={replicas:1}
 	// DecisionEngineServiceTemplate - define the watcher-decision-engine service
 	DecisionEngineServiceTemplate WatcherDecisionEngineTemplate `json:"decisionengineServiceTemplate"`
+
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// APIOverride, provides the ability to override the generated
+	// manifest of several child resources.
+	APIOverride Override `json:"apiOverride,omitempty"`
+}
+
+type Override struct {
+	// +kubebuilder:validation:Optional
+	// Route overrides to use when creating the public service endpoint
+	Route *route.OverrideSpec `json:"route,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// TLS - overrides tls parameters for public endpoint
+	TLS *TLSServiceOverride `json:"tls,omitempty"`
+}
+
+// TLSServiceOverride overrides tls parameters for public endpoint
+type TLSServiceOverride struct {
+	// +kubebuilder:validation:Optional
+	// Name of a Secret in the same Namespace as the service, containing the server's private key, public certificate
+	// and CA certificate for TLS.
+	// The Secret must store these as tls.key, tls.crt and ca.crt respectively.
+	SecretName string `json:"secretName,omitempty"`
```

Comment:
Which part will we need to remove from CRD when integrating into openstack-operator? this TLS.SecretName or also the RouteOverride part?

Reply:
we'll need to remove the whole Override struct and move it to the watcher section in the openstack operator.